docs/api/python/rnn.md

```diff
@@ -0,0 +1,339 @@
+                                   layout='TNC', \
+                                   merge_outputs=False)
+# In this case, 'outputs' is a list of symbols. Each symbol is of shape (batch_size, hidden_dim).
+```
+
+```eval_rst
+.. note:: Loading and saving models that are built with RNN cells API requires using
+    `mx.rnn.load_rnn_checkpoint`, `mx.rnn.save_rnn_checkpoint`, and `mx.rnn.do_rnn_checkpoint`.
+    The list of all the used cells should be provided as the first argument to those functions.
+```
+
+### Basic RNN cells
+
+`rnn` module supports the following RNN cell types.
+
+```eval_rst
+.. autosummary::
+    :nosignatures:
+
+    LSTMCell
+    GRUCell
+    RNNCell
+```
+
+### Modifier cells
+
+```eval_rst
+.. autosummary::
+    :nosignatures:
+
+    BidirectionalCell
+    DropoutCell
+    ZoneoutCell
+```
+
+A modifier cell takes in one or more cells and transforms the output of those cells.
+`BidirectionalCell` is one example. It takes two cells for forward unroll and backward unroll
+respectively. After unrolling, the outputs of the forward and backward pass are concatenated.
+```python
+# Bidirectional cell takes two RNN cells, for forward and backward pass respectively.
+# Having different types of cells for forward and backward unrolling is allowed.
+bi_cell = mx.rnn.BidirectionalCell(
+                 mx.rnn.LSTMCell(num_hidden=50),
+                 mx.rnn.GRUCell(num_hidden=75))
+outputs, states = bi_cell.unroll(length=sequence_length, \
+                                 inputs=embedded_seq, \
+                                 layout='TNC', \
+                                 merge_outputs=True)
+# The output feature is the concatenation of the forward and backward pass.
+# Thus, the number of output dimensions is the sum of the dimensions of the two cells.
+# 'outputs' is the symbol 'bi_out_output' of shape (sequence_length, batch_size, 125L)
+
+# The states of the BidirectionalCell is a list of two lists, corresponding to the
+# states of the forward and backward cells respectively.
+```
+```eval_rst
+.. note:: BidirectionalCell cannot be called or stepped, because the backward unroll requires the output of
+    future steps, and thus the whole sequence is required.
+```
+
+Dropout and zoneout are effective regularization techniques that can be applied on RNN. `rnn`
+module provides `DropoutCell` and `ZoneoutCell` for regularization on the output and recurrent
+states of RNN. `ZoneoutCell` takes one RNN cell in the constructor, and are unrolled like
+other cells.
+```python
+zoneout_cell = mx.rnn.ZoneoutCell(lstm_cell, zoneout_states=0.5)
+outputs, states = zoneout_cell.unroll(length=sequence_length, \
+                                      inputs=embedded_seq, \
+                                      layout='TNC', \
+                                      merge_outputs=True)
+```
+`DropoutCell` performs dropout on the output of the sequence input. It can be used in a stacked
```

Comment:
"dropout on the output of the sequence input"

basically this says "dropout on the output of the input". What output of the input means is not clear. Can you please rephrase?